### PR TITLE
Fixes #14955 - Show nice message for invalid mac, ip

### PR DESCRIPTION
**Pull request by xprazak2**

### CODE REVIEW

**Comment by timogoebel** [comment]
@xprazak2: Looks good to me. Would you mind to add some test(s) for this?

**Comment by dLobatog** [comment]
@xprazak2 Thanks, nice catch :smile: !

I think it would make sense to wrap these rescue blocks in a method, like `catch_network_error` does [in foreman-docker](https://github.com/theforeman/foreman-docker/blob/9ea405911cef1f02ca42db77c16bea4fe297093e/app/controllers/image_search_controller.rb#L55). In the rescue block you can write some common behavior that's wanted for all of them. Currently they're returning nil, but I think it's fair to say they should also log the error in `logger.debug`.

**Comment by xprazak2** [comment]
I added tests and wrapped the error handling in a method.

**Comment by timogoebel** [comment]
Looks good to me, thanks @xprazak2 . If @dLobatog is fine with this as well, we can go ahead and merge this.

File: test/unit/orchestration/common.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      include Orchestration::Common
+
+      def some_method
+        handle_validation_errors do
+          raise Net::Validations::Error, 'validation error for SomeClass'
+        end
+      end
+    end
+
+    SomeClass.any_instance.stubs(:logger).returns(Rails.logger)
```
Comment (timogoebel) [inline test/unit/orchestration/common.rb]: Why not `delegate :logger, :to => :Rails`?

File: test/unit/orchestration/dhcp_test.rb [hunk 2]
```diff
@@ -237,4 +237,9 @@ def teardown
     Nic::Managed.any_instance.expects(:set_dhcp).raises(StandardError, 'DHCP test failure')
     refute h.interfaces.first.rebuild_dhcp
   end
+
+  test "dhcp_record should return nil for invalid mac" do
+    h = FactoryGirl.build(:host, :with_dhcp_orchestration, :interfaces => [FactoryGirl.build(:nic_primary_and_provision, :mac => "aaaaaa")])
```
Comment (timogoebel) [inline test/unit/orchestration/dhcp_test.rb]: Nitpick: Use `host` here, it better describes the variable than `h`.

**Comment by xprazak2** [comment]
I renamed the variables and changed the stub to delegation.

**Comment by dLobatog** [comment]
Merged as e216302087f27280dc97b8ed11f1082e57ef6545, thanks @xprazak2!

**Pull request closed without merging**